Packages/com.unity.inputsystem/InputSystem/Devices/ScreenKeyboard.cs

```diff
@@ -0,0 +1,298 @@
+        LostFocus
+    }
+
+    // Note: This struct is marshalled to iOS native code, don't change the layout
+    //       Also don't use auto properties, since it messes up the layout
+    /// <summary>
+    /// Describes the appearance of screen keyboard.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct ScreenKeyboardShowParams
+    {
+        private ScreenKeyboardType m_Type;
+        private string m_InitialText;
+        private string m_PlaceholderText;
+        private bool m_Autocorrection;
+        private bool m_Multiline;
+        private bool m_Secure;
+        private bool m_Alert;
+        private bool m_InputFieldHidden;
+
+        /// <summary>
+        /// The type of the keyboard.
+        /// </summary>
+        public ScreenKeyboardType type { get => m_Type; set => m_Type = value; }
+
+        /// <summary>
+        /// The default input field text which is present when keyboard is shown.
+        /// </summary>
+        public string initialText { get => m_InitialText; set => m_InitialText = value; }
+
+        /// <summary>
+        /// Placeholder text when input field text is empty.
+        /// </summary>
+        public string placeholderText { get => m_PlaceholderText; set => m_PlaceholderText = value; }
+
+        /// <summary>
+        /// Is auto correction applied?
+        /// </summary>
+        public bool autocorrection { get => m_Autocorrection; set => m_Autocorrection = value; }
+
+        /// <summary>
+        /// Can more than one line of text be entered in the input field ?
+        /// </summary>
+        public bool multiline { get => m_Multiline; set => m_Multiline = value; }
+
+        /// <summary>
+        /// Is the text masked (for passwords, etc)?
+        /// </summary>
+        public bool secure { get => m_Secure; set => m_Secure = value; }
+
+        /// <summary>
+        /// Is the keyboard opened in alert mode?
+        /// Note: Only available on iOS
+        /// </summary>
+        public bool alert { get => m_Alert; set => m_Alert = value; }
+
+        /// <summary>
+        /// Is input field hidden?
+        /// </summary>
+        public bool inputFieldHidden { get => m_InputFieldHidden; set => m_InputFieldHidden = value; }
+    }
+
+    /// <summary>
+    /// Provides screen keyboard callbacks
+    /// </summary>
+    /// Note: We use delegates here instead of events to disallow a potential bug from users
+    ///       If these would be events, the order in which callbacks would invoked would be undefined.
+    ///       Imagine if in one of the inputFieldTextChanged callbacks we change the input field text.
+    ///       The other inputFieldTextChanged would still receive the original changed text which is wrong and could lead to potential problems.
+    /// Note: We use this approach with class callbacks, since it's more future proof, if we add an additional callback in the future, it won't break existing user projects.
```

Comment:
Is it OK for these comments to be part of doc-comment (three slashes)?

Reply:
since they're outside summary, I don't think the go to docs , not sure @Rene-Damm